### PR TITLE
Sbom syft lib

### PR DESCRIPTION
**Pull request by robdimsdale**
## Summary

This PR replaces the existing SBOM functionality that this buildpack provides (via `cyclonedx-node-module`) with the new functionality introduced in Packit by [this PR](https://github.com/paketo-buildpacks/packit/pull/255/).

We should not merge this PR until the corresponding PR in `packit` has been merged.

Additionally, we should have further discussion about whether this buildpack should exist at all, or whether the BOM generation is better located in other buildpacks in the language family (e.g. `npm-install` and/or `npm-start`).

## Use Cases

This generates SBOM files in both `syft` and `spdx` format as well as the existing `cycloneDX` format.

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: build.go [hunk 1]
```diff
@@ -19,91 +17,55 @@ type DependencyManager interface {
 	GenerateBillOfMaterials(dependencies ...postal.Dependency) []packit.BOMEntry
```
Comment (thitch97) [inline build.go:17]: We should be able to remove this method with these changes

Comment (robdimsdale) [reply build.go:17]: Yes, you're right! There are a few other changes I'd like to make before proposing this for merge. I'd like to wait for the underlying packit library interface to stabilize, and then clean this PR up.

Can you add the `draft` label to this PR? I can't do that myself.

File: build.go [hunk 2]
```diff
@@ -19,91 +17,55 @@ type DependencyManager interface {
...
+		layer.Launch = true
+
+		layer.SBOM.Set("cdx.json", bom.Format(sbom.CycloneDXFormat))
+		layer.SBOM.Set("syft.json", bom.Format(sbom.SyftFormat))
+		layer.SBOM.Set("spdx.json", bom.Format(sbom.SPDXFormat))
+
+		bomEntries := make(packit.SBOMEntries)
+		bomEntries.Set("cdx.json", bom.Format(sbom.CycloneDXFormat))
+		bomEntries.Set("syft.json", bom.Format(sbom.SyftFormat))
+		bomEntries.Set("spdx.json", bom.Format(sbom.SPDXFormat))
```
Comment (ryanmoran) [inline build.go:60]: Why do we have this set of BOM entries? Is this not duplicative of what already is attached to the layer?

Comment (robdimsdale) [reply build.go:60]: No real reason. We were having unrelated issues with the layer not showing up at launch so we added the entries to the launch metadata too.

**Comment by robdimsdale** [comment]
I would like to close this PR with the proposal that we move this logic to inside the `npm-install` and `yarn-install` buildpacks instead.

However, that would implicitly mean that we wouldn't be able to support generating a BOM for projects that don't have a `package-lock.json` (or `yarn.lock`). I _think_ that syft doesn't generate a BOM in this case anyway, but we can validate that if we feel strongly.

Do folks want to see an RFC to move this bom generation logic to the module buildpacks? Or, instead, can we get a 👍 on this comment?

**Comment by ryanmoran** [comment]
> Do folks want to see an RFC to move this bom generation logic to the module buildpacks? Or, instead, can we get a 👍 on this comment?

I don't need to see an RFC. I think moving to the `npm-install` and `yarn-install` buildpacks makes sense. We can further address the "no package manager" case at a later point.

**Comment by fg-j** [comment]
@robdimsdale Closing this since we've determined that syft SBOM functionality will make this buildpack obsolete.

**Pull request closed without merging**